pkg/util/provider/machinecontroller/migrate_machineclass.go

```diff
@@ -382,26 +391,36 @@ func (c *controller) TryMachineClassMigration(classSpec *v1alpha1.ClassSpec) (*v
 	var (
 		err                          error
 		providerSpecificMachineClass interface{}
+		updateMachineClassObjects    = true
 	)
 
 	// Get the provider specific (e.g. AWSMachineClass) from the classSpec
 	if providerSpecificMachineClass, err = c.getProviderSpecificMachineClass(classSpec); err != nil {
-		return nil, nil, machineutils.ShortRetry, err
+		if err.Error() == "ProviderMachineClass not found. Found a MachineClass with matching name" {
```

Comment:
Looks a bit strange to compare the error string this way, wondering if there could be an elegant way, . If not, a string-constant could be used to compare.

Reply:
This is the way I found online on searching as well - https://stackoverflow.com/questions/39121172/how-to-compare-go-errors. But yes, will update it to string constants.